src/main/java/seedu/address/model/ingredient/Ingredient.java

```diff
@@ -0,0 +1,38 @@
+package seedu.address.model.ingredient;
+
+/**
+ * Represents an ingredient in the list of ingredients
+ * Guarantees: TODO
+ */
+public abstract class Ingredient {
+
+    // Identity field
+    protected final IngredientName name;
+
+    // Data fields
+    protected IngredientQuantity quantity;
+    // TODO: ingredient cost
+
+    public Ingredient(IngredientName name, IngredientQuantity quantity) {
```

Comment:
Constructor here only need name if according to user Story User
quantity can set to 0 and use getter and setter to set it
(assuming we have the idea of batch at back of the mind, quantity only increase when new batch is in, and decrease as it is sold out

Reply:
hmmm, this really depends on your definition rather than commands
There is two definition I can think of.. One is multiple instance of Ingredient that with the same name, so each batch come in, we create a new instance of ingredient...(this is make sense when quantity is involved in constructor.(this command is open of stock updater)

Other is  only manager can add in new ingredient. (stock taker updates the quantity). In this case, constructor should not include quantity. As each modification is solved by setter..
 Please clarify which definition for ingredient can are u referring to ?